cegs_portal/search/templates/search/v1/experiment.html

```diff
@@ -348,10 +365,287 @@
+    plot = (values, subtitle) => {
+        return vl
+            .markCircle()
+            .data(values)
+            .width(450)
+            .height(400)
+            .encode(
+                vl.y().fieldQ("0").title("-log10(p-value)"),
+                vl.x().fieldQ(1).title("log2(FC)"), //.scale({"domain":[0, 2], "clamp": true}),
+                vl
+                    .color()
+                    .fieldN(2)
+                    .title("Gene Type")
+                    .legend({labelExpr: "{0: 'Targeting', 1: 'Non-Targeting'}[datum.label]"}),
+                vl.tooltip(vl.fieldN(3))
+            )
+            .title({text: "Volcano Plot", subtitle: subtitle, anchor: "start"})
+            .render()
+
+            .then((viewElement) => {
+                // render returns a promise to a DOM element containing the chart
+                // viewElement.value contains the Vega View object instance
+                document.getElementById("volcano").appendChild(viewElement);
+            });
+    };
+
+    fetch("{% get_static_prefix %}search/experiments/{{ experiment.accession_id }}/vpdata.bin")
+        .then((response) => response.arrayBuffer())
+        .then((buffer) => {
+            let length = buffer.byteLength;
+            let decoder = new TextDecoder();
+            let data = new DataView(buffer);
+            let offset = 0;
+            let values = [];
+            while (offset < length) {
+                let pVal = data.getFloat32(offset);
+                let avgFC = data.getFloat32(offset + 4);
+                let type = data.getUint8(offset + 8);
+                let symLen = data.getUint8(offset + 9);
+                offset += 10;
+                let symArray = new Uint8Array(buffer, offset, symLen);
+                let sym = decoder.decode(symArray);
+                values.push([pVal, avgFC, type, sym]);
+                offset += symLen;
+            }
+            return values;
+        })
+        .then((values) => {
+            return plot(values, "Crop of Non-Significant, Low Effect Observations");
+        });
+
+    //Function to process both Histgrams("Number of Cells per gRNA" and "Number od gRNA per cell" )
+    function processHistogramData(buffer) {
+        let data = new DataView(buffer);
+        let offset = 0;
+        let values = [];
+        let binSize = data.getUint32(offset);
+        let mean = data.getFloat32(offset + 4);
+        let median = data.getFloat32(offset + 8);
+        let binCount = data.getUint32(offset + 12);
+        offset += 16;
+        for (let i = 0; i < binCount; i++, offset += 4) {
+            let binValue = data.getUint32(offset);
+            values.push({
+                bin_start: i * binSize,
+                bin_end: (i + 1) * binSize,
+                count: binValue,
+            });
+        }
+
+        return {binSize, mean, median, values};
+    }
+
+    function createHistogramChart(data, color, chart_title, x_axis_title, elementId) {
```

Comment:
good name!

Reply:
Thanks Thomas!!!! Yes! I probably should have consulted you about the new branch name I just created. Ah!!